src/mixins/map-mixin.js

```diff
@@ -329,6 +342,16 @@ export default function mapMixin(
             ydim.filter([_chart._minCoord[1], _chart._maxCoord[1]])
           }
         }
+        else if(typeof layer.polyDim === "function") {
+          const polyDim = layer.polyDim()
+          if(polyDim !== null) {
+            redrawall = true
+            polyDim.filterST_Intersects([[bounds._sw.lng, bounds._sw.lat],
```

Comment:
@vastcharade Yeah, I agree. I have performance concern on linemap when using either ST_Contains or ST_Intersects as well. What would be a less expensive query to filter on bounds? I haven't actually tried to compare the one you mentioned in Slack because of the caveat to have false positives.